src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectContext.cs

```diff
@@ -7,26 +7,34 @@
 namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
 {
     /// <summary>
-    /// Project context to initialize properties and items of a Workspace project created with <see cref="IProjectContextFactory.CreateProjectContext(string, string, string, Guid, string, Shell.Interop.IVsHierarchy, CommandLineArguments)"/>. 
+    /// Project context to initialize properties and items of a Workspace project created with <see cref="IProjectContextFactory.CreateProjectContext(string, string, string, Shell.Interop.IVsHierarchy)"/>. 
     /// </summary>
-    internal interface IProjectContext
+    internal interface IProjectContext : IDisposable
     {
         // Project properties.
-        ProjectId Id { get; }
         string DisplayName { get; set; }
         string ProjectFilePath { get; set; }
         Guid Guid { get; set; }
         string ProjectType { get; set; }
+        bool DesignTimeBuildStatus { get; set; }
```

Comment:
Can we give a name that indicates what it means? "status" is not something I would associate with a bool. Secondly, I'm not sure I can set it correctly. To even instantiate this object, I need to pass you a command-line - I can't even instantiate the language service is the design-time build fails.